docs/MAINTAINERS.md

```diff
@@ -1,52 +1,180 @@
 ```
 
-Upgrading Electron
-------------------
+- Open a pull request against `master` titled "Release v1.0.0"
+
+### Generating binaries
+
+**Environment**
+
+Make sure to set the analytics tokens when generating production release binaries:
+
+```bash
+export ANALYTICS_SENTRY_TOKEN="xxxxxx"
+export ANALYTICS_MIXPANEL_TOKEN="xxxxxx"
+```
+
+#### Linux
+
+##### Clean dist folder
+
+**NOTE:** Make sure to adjust the path as necessary (here the Etcher repository has been cloned to `/home/$USER/code/etcher`)
+
+```bash
+docker run -v "/home/$USER:/home/$USER" "busybox rm -rf /home/$USER/code/etcher/release /home/$USER/code/etcher/node_modules /home/$USER/code/etcher/build /home/$USER/code/etcher/dist"
+```
+
+##### Generating artefacts
+
+```bash
+# x64
+
+# Build Debian packages
+./scripts/build/docker/run-command.sh -r x64 -s . -c "make electron-develop && make RELEASE_TYPE=production electron-installer-debian"
+# Build RPM packages
+./scripts/build/docker/run-command.sh -r x64 -s . -c "make electron-develop && make RELEASE_TYPE=production electron-installer-redhat"
+# Build AppImages
+./scripts/build/docker/run-command.sh -r x64 -s . -c "make electron-develop && make RELEASE_TYPE=production electron-installer-appimage"
+# Build CLI
+./scripts/build/docker/run-command.sh -r x64 -s . -c "make electron-develop && make RELEASE_TYPE=production cli-installer-tar-gz"
+
+# x86
+
+# Build Debian packages
+./scripts/build/docker/run-command.sh -r x86 -s . -c "make electron-develop && make RELEASE_TYPE=production electron-installer-debian"
+# Build RPM packages
+./scripts/build/docker/run-command.sh -r x86 -s . -c "make electron-develop && make RELEASE_TYPE=production electron-installer-redhat"
+# Build AppImages
+./scripts/build/docker/run-command.sh -r x86 -s . -c "make electron-develop && make RELEASE_TYPE=production electron-installer-appimage"
+# Build CLI
+./scripts/build/docker/run-command.sh -r x86 -s . -c "make electron-develop && make RELEASE_TYPE=production cli-installer-tar-gz"
+```
 
-- Upgrade the `electron` dependency version in `package.json` to an *exact
-  version* (no `~`, `^`, etc).
+#### Mac OS
+
+**ATTENTION:** For production releases you'll need the code-signing key,
+and set `CSC_NAME`, `CSC_LINK`, and `CSC_KEY_PASSWORD` to generate signed binaries on Mac OS.
+
+**NOTE:** The CLI is not code-signed for either at this time.
+
+```bash
+make electron-develop
+
+# Build the CLI
+make RELEASE_TYPE=production cli-installer-tar-gz
+# Build the zip
+make RELEASE_TYPE=production electron-installer-app-zip
+# Build the dmg
+make RELEASE_TYPE=production electron-installer-dmg
+```
+
+#### Windows
+
+**ATTENTION:** For production releases you'll need the code-signing key,
+and set `CSC_NAME`, `CSC_LINK`, and `CSC_KEY_PASSWORD` to generate signed binaries on Windows.
+
+**NOTE:** The CLI is not code-signed for either at this time.
+
+```bash
+make electron-develop
+
+# Build the CLI
+make RELEASE_TYPE=production cli-installer-zip
```

Comment:
You should clarify that `TARGET_ARCH` is needed to build x86 packages.

Reply:
Done